instrumentation/opentelemetry-instrumentation-logging/src/opentelemetry/instrumentation/logging/__init__.py

```diff
@@ -0,0 +1,115 @@
+    OTEL_PYTHON_LOG_CORRELATION,
+    OTEL_PYTHON_LOG_FORMAT,
+    OTEL_PYTHON_LOG_LEVEL,
+)
+from opentelemetry.trace import (
+    INVALID_SPAN,
+    INVALID_SPAN_CONTEXT,
+    get_current_span,
+    get_tracer_provider,
+)
+
+__doc__ = _MODULE_DOC
+
+LEVELS = {
+    "debug": logging.DEBUG,
+    "info": logging.INFO,
+    "warning": logging.WARNING,
+    "error": logging.ERROR,
+}
+
+
+class LoggingInstrumentor(BaseInstrumentor):  # pylint: disable=empty-docstring
+    __doc__ = """An instrumentor for stdlib logging module.
+
+    This instrumentor injects tracing context into logging records and optionally sets the global logging format to the following:
+
+    .. code-block::
+
+        {default_logging_format}
+
+    See `BaseInstrumentor`
+    """.format(
+        default_logging_format=DEFAULT_LOGGING_FORMAT
+    )
+
+    _factory_registered = False
+    _is_instrumented = False
+
+    def _instrument(self, **kwargs):
+        LoggingInstrumentor._is_instrumented = True
+        service_name = ""
+        provider = kwargs.get("tracer_provider", None) or get_tracer_provider()
+        resource = provider.resource if provider else None
+        if resource:
+            service_name = resource.attributes.get("service.name")
+
+        old_factory = logging.getLogRecordFactory()
+        if not LoggingInstrumentor._factory_registered:
+            LoggingInstrumentor._factory_registered = True
+
+            def record_factory(*args, **kwargs):
+                record = old_factory(*args, **kwargs)
+                if not LoggingInstrumentor._is_instrumented:
+                    return record
+
+                record.otelSpanID = ""
+                record.otelTraceID = ""
+                record.otelServiceName = service_name
+
+                span = get_current_span()
+                if span != INVALID_SPAN:
+                    ctx = span.get_span_context()
+                    if ctx != INVALID_SPAN_CONTEXT:
+                        record.otelSpanID = format(ctx.span_id, "016x")
+                        record.otelTraceID = format(ctx.trace_id, "032x")
+                return record
+
+            logging.setLogRecordFactory(record_factory)
+
+        set_logging_format = kwargs.get(
+            "set_logging_format",
+            environ.get(OTEL_PYTHON_LOG_CORRELATION, "false").lower()
+            == "true",
+        )
+
+        if set_logging_format:
+            log_format = kwargs.get(
+                "logging_format", environ.get(OTEL_PYTHON_LOG_FORMAT, None)
+            )
+            log_format = log_format or DEFAULT_LOGGING_FORMAT
+
+            log_level = kwargs.get(
+                "log_level", LEVELS.get(environ.get(OTEL_PYTHON_LOG_LEVEL))
+            )
+            log_level = log_level or logging.INFO
+
+            logging.basicConfig(format=log_format, level=log_level)
+
+    def _uninstrument(self, **kwargs):
+        LoggingInstrumentor._is_instrumented = False
```

Comment:
Would keeping the `old_factory` and resetting it here be easier than tracking the `_is_instrumented` and `_factory_registered` booleans?